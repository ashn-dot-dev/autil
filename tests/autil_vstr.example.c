#define AUTIL_IMPLEMENTATION
#include "../autil.h"
#include "test.h"

static struct autil_vstr const static_vstr = AUTIL_VSTR_INIT_STR_LITERAL("foo");

int
main(void)
{
    printf(
        "static literal \"%.*s\": %zu bytes\n",
        (int)static_vstr.count,
        static_vstr.start,
        static_vstr.count);

    struct autil_vstr local_vstr = AUTIL_VSTR_INIT_STR_LITERAL("bar");
    printf(
        "local literal \"%.*s\": %zu bytes\n",
        (int)local_vstr.count,
        local_vstr.start,
        local_vstr.count);
    local_vstr = (struct autil_vstr)AUTIL_VSTR_INIT_STR_LITERAL("baz");
    printf(
        "reassigned local literal \"%.*s\": %zu bytes\n",
        (int)local_vstr.count,
        local_vstr.start,
        local_vstr.count);

    char const* const str = "the string foobar";
    struct autil_vstr init = {&str[11], 6};
    printf("\"%.*s\": %zu bytes\n", (int)init.count, init.start, init.count);

    return EXIT_SUCCESS;
}
