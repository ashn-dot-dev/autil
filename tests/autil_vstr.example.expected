static literal "foo": 3 bytes
local literal "bar": 3 bytes
reassigned local literal "baz": 3 bytes
"foobar": 6 bytes
